File: simple-stack-example-mvvm-fragments/src/main/java/com/example/android/architecture/blueprints/todoapp/application/BackstackHolder.java
package com.example.android.architecture.blueprints.todoapp.application;

import com.zhuinden.simplestack.Backstack;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Owner on 2017. 07. 26..
 */

@Singleton
public class BackstackHolder {
    private Backstack backstack;

    @Inject
    public BackstackHolder() {
    }

    public Backstack getBackstack() {
        return this.backstack;
    }

    public void setBackstack(Backstack backstack) {
        this.backstack = backstack;
    }
}

File: samples/legacy-architecture-samples/simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/util/BackstackHolder.kt
package com.zhuinden.simplestackdemoexamplemvp.util

import com.zhuinden.simplestack.Backstack

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Owner on 2017. 01. 27..
 */
@Singleton
@Deprecated(message = "Singleton outlives the Activity")
class BackstackHolder @Inject constructor() {
    lateinit var backstack: Backstack
}
